airflow/providers/amazon/aws/operators/redshift_data.py

```diff
@@ -0,0 +1,151 @@
+    """
+    Executes SQL Statements against an Amazon Redshift cluster using Redshift Data
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:RedshiftDataOperator`
+
+    :param cluster_identifier: unique identifier of a cluster
+    :param database: the name of the database
+    :param sql: the SQL statement text to run
+    :param db_user: the database user name
+    :param parameters: the parameters for the SQL statement
+    :param secret_arn: the name or ARN of the secret that enables db access
+    :param statement_name: the name of the SQL statement
+    :param with_event: indicates whether to send an event to EventBridge
+    :param timeout: how long in seconds to wait for a response, if 0 don't wait
+    :param poll_interval: how often in seconds to check the query status
+    """
+
+    template_fields = (
+        'cluster_identifier',
+        'database',
+        'sql',
+        'db_user',
+        'parameters',
+    )
+    template_ext = ('.sql',)
+    template_fields_renderers = {'sql': 'sql'}
+
+    def __init__(
+        self,
+        database: str,
+        sql: str,
+        cluster_identifier: Optional[str] = "",
+        db_user: Optional[str] = "",
+        parameters: Optional[list] = None,
+        secret_arn: Optional[str] = "",
+        statement_name: Optional[str] = "",
+        with_event: bool = False,
+        timeout: int = 300,
+        poll_interval: int = 10,
+        aws_conn_id: str = 'aws_default',
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.aws_conn_id = aws_conn_id
+        self.sql = sql
+        self.parameters = parameters
+        self.cluster_identifier = cluster_identifier
+        self.database = database
+        self.db_user = db_user
+        self.secret_arn = secret_arn
+        self.statement_name = statement_name
+        self.with_event = with_event
+        self.timeout = timeout
+        self.statement_id = None
+        if poll_interval > 0:
+            self.poll_interval = poll_interval
+        else:
+            self.log.warning(
+                "Invalid poll_interval:",
+                poll_interval,
+            )
+
+    @cached_property
+    def hook(self) -> RedshiftDataHook:
+        """Create and return an RedshiftDataHook."""
+        return RedshiftDataHook(aws_conn_id=self.aws_conn_id)
+
+    def execute_query(self):
+        resp = self.hook.execute_statement(
+            cluster_identifier=self.cluster_identifier,
+            database=self.database,
+            db_user=self.db_user,
+            sql=self.sql,
+            parameters=self.parameters,
+            secret_arn=self.secret_arn,
+            statement_name=self.statement_name,
+            with_event=self.with_event,
+        )
+        return resp['Id']
+
+    def wait_for_results(self, statement_id):
+        elapsed = 0
+        while elapsed < self.timeout:
```

Comment:
can lean on execution_timeout
```suggestion
        while True:
```